src/tags/tests/test_models.py

```diff
@@ -0,0 +1,276 @@
+from django_tenants.test.cases import TenantTestCase
+from django.contrib.auth import get_user_model
+
+from model_bakery import baker
+
+from tags.models import total_xp_by_tags
+from siteconfig.models import SiteConfig
+from quest_manager.models import Quest, QuestSubmission
+from badges.models import Badge, BadgeAssertion
+
+import itertools
+
+User = get_user_model()
+
+
+class Tag_total_xp_by_tags_Tests(TenantTestCase):
+    """ 
+        Specialized TestClass for testing total_xp_by_tags function
+    """ 
+
+    def setUp(self):
+        self.user = baker.make(User)
+
+        # pre determined amount of qs or ba linked to quest/badges
```

Comment:
Qs or ba? Don't use shortened forms unless they are commonly used through the project. Qs = queryset.

Full variable names please.